### PR TITLE
Fix SelectableText only supports TextSpan exception for Flutter 2

### PR DESCRIPTION
**Pull request by MohamedEL-Torky**
Not much changed, Thanks for your hard work on this package anyway <3

### CODE REVIEW

**Comment by bsneider** [comment]
@Cretezy will this be merged?

**Comment by pr-Mais** [comment]
Thanks for this fix, I had to use this PR's fork in my project to solve this exception, waiting for it to be merged.

File: lib/flutter_linkify.dart [hunk 1]
```diff
@@ -303,20 +308,6 @@ class SelectableLinkify extends StatelessWidget {
   }
 }
 
-class LinkableSpan extends WidgetSpan {
-  LinkableSpan({
-    required MouseCursor mouseCursor,
-    required InlineSpan inlineSpan,
-  }) : super(
-          child: MouseRegion(
```
Comment (Cretezy) [inline lib/flutter_linkify.dart:311]: It seems like we would like to keep `MouseRegion` support for the non-selecting use-case.

I think having `SelectableLinkify` pass an argument to `buildTextSpan` to not use `MouseRegion` would be a better implementation.

**Review by Cretezy — CHANGES_REQUESTED** [review]

File: lib/flutter_linkify.dart (hunk 1 above)
Comment (MohamedEL-Torky) [reply lib/flutter_linkify.dart:311]: @Cretezy 
Unfortunately adding `LinkableSpan` class as a wrapper to support `MouseRegion` will result in throwing the following error for anyone who uses flutter 2 
`"SelectableText only supports TextSpan; Other types of InlineSpan is not allowed"`

Prior Flutter 2 the selectable_text.dart class was declaring the rich text inside the class as dynamic type, but after flutter 2 the type was enforced to be TextSpan that's why the assertion is throwing an error.

on line 230 inside selectable_text.dart file after Flutter 2.0

```
  const SelectableText.rich(
    TextSpan this.textSpan, {
    Key? key,
    this.focusNode,
    this.style,
    this.strutStyle,
```

on line 253 inside selectable_text.dart file before Flutter 2 

```
  const SelectableText.rich(
    this.textSpan, {
    Key key,
    this.focusNode,
    this.style,
    this.strutStyle,
```

So we can't use WidgetSpan to enable using the MouseRegion region property.

There is a merge request on Flutter repo for letting the TextSpan widget support mouseCursor and it was accepted but it was merged on the master channel only.

https://github.com/flutter/flutter/pull/77754

Comment (Cretezy) [reply lib/flutter_linkify.dart:311]: I believe you misunderstood me.

When rendering `Linkify`, we can use mouse regions, therefore we should. When rendering `SelectableLinkify`, we can't use mouse regions, therefore we shouldn't.

The argument should toggle between using mouse regions and not. `Linkify` should have it as `true` while `SelectableLinkify` should have it as `false`.

Comment (MohamedEL-Torky) [reply lib/flutter_linkify.dart]: @Cretezy 

Sorry for the misunderstanding.
I have updated the pull request, please review the updates.

**Review by Cretezy — APPROVED** [review]

**Comment by Cretezy** [comment]
Thank you, just released 5.0.2 with this change!

**Pull request merged**